physics/GFS_MP_generic.F90

```diff
@@ -42,12 +42,24 @@ subroutine GFS_MP_generic_pre_run(im, levs, ldiag3d, do_aw, ntcw, nncl, ntrac, g
         do k=1,levs
           do i=1,im
             save_t(i,k) = gt0(i,k)
-            save_q(1:im,:,1) = gq0(1:im,:,1)
           enddo
         enddo
-        do n=ntcw,ntcw+nncl-1
-          save_q(1:im,:,n) = gq0(1:im,:,n)
-        enddo
+        if(qdiag3d) then
+           do k=1,levs
+              do i=1,im
+                 ! Here, gq0(...,1) is used instead of gq0_water_vapor
+                 ! to be consistent with the GFS_MP_generic_post_run
+                 ! code.
+                 save_qv(i,k) = gq0(i,k,1)
+              enddo
+           enddo
+        endif
+        if(do_aw) then
```

Comment:
I'm curious why this special treatment for do_aw...

Reply:
I don't know what do_aw is for, so I left its implementation unmodified as much as possible.